var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -160,6 +160,8 @@ def yes_no(varstr):
             'PREFIX=%s' % prefix,
             'MFEM_USE_MEMALLOC=YES',
             'MFEM_DEBUG=%s' % yes_no('+debug'),
+            'CXX=%s' % env['CXX'],
+            'MPICXX=%s' % spec['mpi'].mpicxx,
```

Comment:
you need to do 
```
if '+mpi' in spec:
  // set MPICXX
```
your current code will not work you somebody tries to build with `~mpi`.

Reply:
That answers my question, thanks.